kratos/containers/pointer_vector_set.h

```diff
@@ -1072,44 +1083,43 @@ class PointerVectorSet final
                 mData.push_back(TPointerType(&GetReference(it)));
             }
         } else {
-            // first find the largest range
             const auto lower_bound_first = std::lower_bound(mData.begin(), mData.end(), KeyOf(GetReference(first)), CompareKey());
-            const auto upper_bound_last = std::upper_bound(lower_bound_first, mData.end(), KeyOf(GetReference(last-1)), CompareKey());
 
-            // then find the compact sub range
-            const auto upper_bound_first = std::upper_bound(lower_bound_first, upper_bound_last, KeyOf(GetReference(first)), CompareKey());
-            const auto lower_bound_last = std::lower_bound(lower_bound_first, upper_bound_last, KeyOf(GetReference(last-1)), CompareKey());
-
-            if (lower_bound_first == lower_bound_last &&
-                lower_bound_first == upper_bound_first &&
-                lower_bound_first == upper_bound_last)
-            {
-                // all 4 bounds are equal, hence this can be inserted without checking further
+            if (lower_bound_first == mData.end()) {
+                // all are pointing to the end of the vector, hence pushing back.
                 mData.reserve(mData.size() + std::distance(first, last));
-                if (lower_bound_first == mData.end()) {
-                    for (auto it = first; it != last; ++it) {
-                        mData.push_back(TPointerType(&GetReference(it)));
-                    }
-                } else {
-                    // now if the capacity of the new mData is larger than the existing
-                    // capacity, then the current lower_bound_first is invalidated.
-                    // hence needs to find it again.
-                    const auto new_lower_bound = std::lower_bound(mData.begin(), mData.end(), KeyOf(GetReference(first)), CompareKey());
-                    auto current_pos = new_lower_bound - 1;
-                    for (auto it = first; it != last; ++it) {
-                        current_pos = mData.insert(current_pos + 1, TPointerType(&GetReference(it)));
-                    }
+                for (auto it = first; it != last; ++it) {
+                    mData.push_back(TPointerType(&GetReference(it)));
```

Comment:
ok,we just saw this together with @sunethwarna. It only works with intrusive_ptr and it is an error otherwise